### DISCUSSION TITLE
New Features that are crucial for a released game using ALS Refactored

### DISCUSSION THREAD

**Discussion by xkamronx** [Sixze/ALS-Refactored#464] [Ideas]
I have a released game that uses ALS Refactored, and below are a few suggestions that any developer looking at using this in a shipping project need to have added. Most of these are from A LOT of user feedback with the system in general.

1. Exposing FOV to be changeable during runtime. I can't tell you how many users have requested an FOV slider in the settings, but it isn't really possible out of the box.
2. Motion Sickness Setting - Basically an option that would HEAVILY REDUCE or even completely disable the camera bob. As a player myself who doesn't have motion sickness issues in games, I never noticed, but players with medical conditions where motion sickness is an issue, the constant camera bob / sway / etc is unplayable according to feedback from users.
3. Not as crucial, but the ability for the user to modify camera position during runtime has been requested A LOT. Moving the camera closer to their player, further left/right, etc.

Some general feature additions from the development side:
1. Exposing something to BP that allows disabling rolling/mantle/etc for gameplay reasons like being out of stamina.
2. Fixing the large issue with the camera being able to clip in to / through geometry in many situations.

Most of these are likely fairly easy for a C++ developer to handle, but a lot of users of ALS Refactored aren't exactly C++ experts (like myself) so digging through C++ and making changes can be daunting / worrying that something will break.

**Comment by WenzhiShao** [comment]
you might need state machine to limit the ability of player like not rolling when being out of stamina, not to handle it through this system directly
